packages/imperative/src/io/src/IO.ts

```diff
@@ -264,17 +264,41 @@ export class IO {
      * @returns {string} - input with removed newlines
      * @memberof IO
      */
-    public static processNewlines(original: string, lastByte?: number): string {
+    public static processNewlines<T extends string | Buffer>(original: T, lastByte?: number): T {
         ImperativeExpect.toNotBeNullOrUndefined(original, "Required parameter 'original' must not be null or undefined");
+
         if (os.platform() !== IO.OS_WIN32) {
             return original;
         }
-        // otherwise, we're on windows
-        const processed = original.replace(/([^\r])\n/g, "$1\r\n");
-        if ((lastByte == null || lastByte !== "\r".charCodeAt(0)) && original.startsWith("\n")) {
-            return "\r" + processed;
+
+        let processed: T;
+
+        if (typeof original === "string") {
+            processed = original.replace(/([^\r])\n/g, "$1\r\n") as T;
+            if ((lastByte == null || lastByte !== "\r".charCodeAt(0)) && original.startsWith("\n")) {
+                return ("\r" + processed) as T;
+            }
+            return processed;
+        } else {
+            // Create worst-case scenario sized buffer
+            const bufferList = Buffer.alloc(original.length * 2);
+            let bufferIndex = 0;
+            let prevByte = lastByte;
+            for (let i = 0; i < original.length; i++) {
+                const currentByte = original[i];
+                //Check if previous byte is not Carriage Return (13) and if current byte is Line Feed (10)
+                // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                if (prevByte !== 13 && currentByte === 10) {
+                    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                    bufferList[bufferIndex++] = 13;
+                }
+                bufferList[bufferIndex++] = currentByte;
+                prevByte = currentByte;
+            }
+            // Create a buffer slice of the actual used portion
+            processed = bufferList.slice(0, bufferIndex) as T;
+            return processed;
```

Comment:
It's more efficient for `bufferList` to be a list of numbers (`number[]`) rather than a `Buffer`. I think you were initially doing it this way and I may have misled you, sorry about that 😿

Tested performance of calling `IO.processNewLines` 1M times:
```javascript
const testBuf = Buffer.from("abc\ndef\nghi\njkl\nmno\npqr\nstu\nvwx\nyz\0\0");
console.time("test");
for (let i = 0; i < 1000000; i++) {
    const convBuf = IO.processNewLines(testBuf);
}
console.timeEnd("test");
```

| Implementation | Time (ms) |
| --- | --- |
| Original (converted buffer to string) | 545 |
| Current (using `Buffer.alloc`) | 506 |
| Suggested (see below) | 269 |

```suggestion
            const bufferList: number[] = [];
            let prevByte = lastByte;
            for (let i = 0; i < original.length; i++) {
                const currentByte = original[i];
                //Check if previous byte is not Carriage Return (13) and if current byte is Line Feed (10)
                // eslint-disable-next-line @typescript-eslint/no-magic-numbers
                if (prevByte !== 13 && currentByte === 10) {
                    // eslint-disable-next-line @typescript-eslint/no-magic-numbers
                    bufferList.push(13);
                }
                bufferList.push(currentByte);
                prevByte = currentByte;
            }
            processed = Buffer.from(bufferList);
            return processed;
```